chain/client/src/client.rs

```diff
@@ -2004,4 +2012,63 @@ impl Client {
             next_blocks_log.join("\n")
         ))
     }
+
+    pub fn detailed_upcoming_blocks_info_as_web(&self) -> ChunkInfoView {
+        let height_status_map = match self.detailed_upcoming_blocks_info() {
+            Ok(m) => m,
+            Err(_) => HashMap::new(),
+        };
+
+        let next_blocks_log = height_status_map
+            .keys()
+            .sorted()
+            .map(|height| {
+                let val = height_status_map.get(height).unwrap();
+
+                let block_debug = val
+                    .iter()
+                    .map(|entry| {
+                        let block_info = entry.1;
+                        let chunk_status = block_info
+                            .chunk_hashes
+                            .iter()
+                            .map(|it| {
+                                if block_info.chunks_completed.contains(it) {
+                                    "(done)"
+                                } else if block_info.chunks_received.contains(it) {
+                                    "(received)"
+                                } else if block_info.chunks_requested.contains(it) {
+                                    "(requested)"
+                                } else {
+                                    "."
+                                }
+                            })
+                            .collect::<Vec<&str>>();
```

Comment:
We are using `itertools` here anyway, so you can avoid itermediate `Vec` and join to string directly:

https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.join

Reply:
Good catch!